metagraph/src/graph/representation/succinct/boss_chunk_construct.cpp

```diff
@@ -473,20 +470,22 @@ generate_dummy_1_kmers(size_t k,
             }
             // lift all and reset the first character to the sentinel 0 (apply mask)
             dummy_l1_chunks[F].add(kmer::transform<KMER>(dummy_source, k + 1) + kmer_delta);
-            num_source++;
         }
         // handle leftover sink_gen_it
         while (!sink_gen_it.empty()) {
             KMER_REAL v(sink_gen_it.pop());
             skip_same_suffix(v, sink_gen_it, 1);
             dummy_sink_chunks[F].add(kmer::get_sink_and_lift<KMER>(v, k + 1));
-            num_sink++;
         }
     }
 
+    uint64_t num_sink = 0;
+    uint64_t num_source = 0;
     for (TAlphabet i = 0; i < alphabet_size; ++i) {
         dummy_sink_chunks[i].finish();
         dummy_l1_chunks[i].finish();
+        num_sink += dummy_sink_chunks[i].size();
```

Comment:
I think now it's a bit inconsistent for dummy sink k-mers (the redundant dummy sink k-mers are skipped in chunks but not in counts)

Reply:
I don't understand what you mean. Can you elaborate?

Reply:
never mind, all good